### PR TITLE
Increase unit test coverage for simtools/configuration/configurator.py

### PR DESCRIPTION
**Pull request by GernotMaier**
Test coverage for `simtools/configuration/configurator.py` was with 72% a bit low for such a central module.

Added and improved unit tests.

Fixes also bug in reading of astro.units.Quantity from configuration config, see issue #706

### CODE REVIEW

**Comment by GernotMaier** [comment]
Fixed the issue with the integration tests now.  Requires some changes to the configurator to check also for strings which can be converted to astropy.units.Quantities. Added also unit tests to cover these cases.

@orelgueta - this is ready for review now.

File: simtools/configuration/commandline_parser.py [hunk 1]
```diff
@@ -410,6 +413,14 @@ def azimuth_angle(angle):
         except ValueError:
             logger.debug(
                 "The azimuth angle provided is not a valid numeric value. "
+                "Will check if it is an astropy.Quantity instead"
+            )
+        try:
+            fangle = u.Quantity(angle).to("deg")
+            return fangle.to("deg")
+        except TypeError:
```
Comment (orelgueta) [inline simtools/configuration/commandline_parser.py:421]: There are (at least) two other exceptions that can be raised here. One is the case where the user provides a valid quantity but it cannot be converted to deg (e.g., kg), and the other is when the unit provided is not even a valid unit (e.g., "90 north").
Technically an error will be raised that will provide some details. However, perhaps you want to provide an even friendlier error message (far from critical though).

File: simtools/configuration/commandline_parser.py [hunk 2]
```diff
@@ -360,17 +360,20 @@ def zenith_angle(angle):
         logger = logging.getLogger(__name__)
 
         try:
-            fangle = float(angle)
+            fangle = float(angle) * u.deg
         except ValueError:
-            logger.error("The zenith angle provided is not a valid numeric value.")
-            raise
-        if fangle < 0.0 or fangle > 180.0:
+            fangle = u.Quantity(angle).to("deg")
```
[the remark this answers is not in the document]
Comment (GernotMaier) [reply simtools/configuration/commandline_parser.py:365]: Added.

File: simtools/configuration/commandline_parser.py [hunk 3]
```diff
@@ -410,6 +413,14 @@ def azimuth_angle(angle):
         except ValueError:
             logger.debug(
                 "The azimuth angle provided is not a valid numeric value. "
+                "Will check if it is an astropy.Quantity instead"
+            )
+        try:
+            fangle = u.Quantity(angle).to("deg")
+            return fangle.to("deg")
```
[the remark this answers is not in the document]
Comment (GernotMaier) [reply simtools/configuration/commandline_parser.py:420]: Fixed.

File: simtools/configuration/commandline_parser.py (hunk 1 above)
Comment (GernotMaier) [reply simtools/configuration/commandline_parser.py:421]: I do think the error message `astropy.units.core.UnitConversionError: 'kg' (mass) and 'deg' (angle) are not convertible` is very clear and hard to improve. Also think that `ValueError: 'north' did not parse as unit` is quite clear and difficult to improve. Suggest not too add additional exceptions.

**Comment by GernotMaier** [comment]
Thanks @orelgueta 

Fixed the issues and commented on the other. Not sure about the test, in my view this is covered.

File: simtools/configuration/commandline_parser.py [hunk 4]
```diff
@@ -360,17 +360,20 @@ def zenith_angle(angle):
...
 
         try:
-            fangle = float(angle)
+            fangle = float(angle) * u.deg
         except ValueError:
-            logger.error("The zenith angle provided is not a valid numeric value.")
-            raise
-        if fangle < 0.0 or fangle > 180.0:
+            fangle = u.Quantity(angle).to("deg")
+        except TypeError:
```
[the remark this answers is not in the document]
Comment (orelgueta) [reply simtools/configuration/commandline_parser.py:366]: But you don't actually raise the TypeError (which I think you should). I am not sure how that test passes then though.

[the remark this answers is not in the document]
Comment (GernotMaier) [reply simtools/configuration/commandline_parser.py]: Ah! The TypeError was raised inside the `except ValueError`. Fixed this.

**Review by orelgueta — APPROVED** [review]
Once all tests pass, we can merge

**Pull request merged**